cpp/src/arrow/ipc/ipc-memory-test.cc

```diff
@@ -0,0 +1,81 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <gtest/gtest.h>
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "arrow/ipc/memory.h"
+#include "arrow/ipc/test-common.h"
+#include "arrow/test-util.h"
+#include "arrow/util/buffer.h"
+#include "arrow/util/status.h"
+
+namespace arrow {
+namespace ipc {
+
+class TestMemoryMappedSource : public ::testing::Test, public MemoryMapFixture {
+ public:
+  void TearDown() {
+    MemoryMapFixture::TearDown();
+  }
+};
+
+TEST_F(TestMemoryMappedSource, InvalidUsages) {
+}
+
+TEST_F(TestMemoryMappedSource, WriteRead) {
+  int64_t buffer_size = 1024;
+  std::vector<uint8_t> buffer(buffer_size);
+
+  test::random_bytes(1024, 0, &buffer[0]);
```

Comment:
Old habits die hard -- I'm gonna try to get in the habit of using `data()` in C++1y